### PR TITLE
Speed up CI runtime

### PR DESCRIPTION
**Pull request by datumbox**
This PR contains a number of improvements that aim to speed up the building and testing times of the project. None of the changes alter the quality and quantity of tests that we do (aka we don't modify batch sizes, vector sizes, iterations, number of tests etc). Further improvements can be achieved if we make such modifications on the slowest tests.

See sub-PRs for more details:
- Speedup test_ucf101 (#3187): 40% improvement
- Speedup Cmake builds (#3186): 15-50% improvement depending on platform
- Speedup test_autoaugment (#3190): 55% improvement
- Speedup DeformConvTester (#3191): 7-12% improvement depending on the platform
- Speedup InceptionV3 and GoogleNet on Windows (#3196): 96% improvement

To properly estimate the CI cost savings and dev time improvements of all the above changes we will require doing multiple runs which is both impractical and costly. Instead the stats provided above come from a couple of runs and thus should be taken with a grain of salt. At any case to get a rough idea of the impact of the improvements:
- **CI Costs:** Given that the test in the CI jobs run serially, with a "back-of-the-envelope" calculation (factoring in the tests across all platforms, python versions and devices) we save about 70 minutes of total execution time (see sub-PRs for more info). This directly affects the CI costs, though it's not easy to get a dollar figure due to differences in pricing on various instances.
- **Development time:** Since the CI jobs across platforms run in parallel, to estimate the improvements on the waiting times for reviewing/merging PRs, we focus only on Windows GPU which is the slowest CI job. By comparing the [before](https://app.circleci.com/pipelines/github/pytorch/vision/5546/workflows/66cfb30c-8c78-40eb-af95-ad9f01b6930f/jobs/350618) and [after](https://app.circleci.com/pipelines/github/pytorch/vision/5598/workflows/ad7b6782-4dfb-4806-bc95-c4864c312533/jobs/353679), we get a rough improvement of about 45% or 22 minutes less waiting.

### CODE REVIEW

**Comment by datumbox** [comment]
The failing tests are not related to this PR.

File: test/test_ops.py [hunk 1]
```diff
@@ -496,6 +497,7 @@ def expected_fn(self, x, weight, offset, mask, bias, stride=1, padding=0, dilati
         out += bias.view(1, n_out_channels, 1, 1)
         return out
 
+    @lru_cache(maxsize=None)
```
Comment (vfdev-5) [inline test/test_ops.py:500]: Deciding on caching random data is not that simple IMO. There are pros/cons of that...

File: packaging/windows/internal/build_cmake.bat [hunk 2]
```diff
@@ -1,3 +1,3 @@
 @echo on
-msbuild "-p:Configuration=Release" torchvision.vcxproj
-msbuild "-p:Configuration=Release" INSTALL.vcxproj
+msbuild "-p:Configuration=Release" "-p:BuildInParallel=true" "-p:MultiProcessorCompilation=true" "-p:CL_MPCount=%1" torchvision.vcxproj -maxcpucount:%1
```
Comment (vfdev-5) [inline packaging/windows/internal/build_cmake.bat:2]: So, here `/MP` is not the right option ? Funny that on windows we have to set 2 more flags...

**Review by vfdev-5 — COMMENTED** [review]
Few comments :) 
Nice description of the PR @datumbox !

File: test/test_ops.py (hunk 1 above)
Comment (datumbox) [reply test/test_ops.py:500]: I agree there are pros/cons. Note that this is something we do on tests only and seems to help us speed-wise. I don't have a strong opinion about keeping it if it causes issues.

File: packaging/windows/internal/build_cmake.bat (hunk 2 above)
Comment (datumbox) [reply packaging/windows/internal/build_cmake.bat:2]: As far as I understand after reading the Microsoft docs, `MultiProcessorCompilation` == `/MP`. And there are many more flags, related to whether the parallelism is across projects or across C++ files. Frankly I can't say I'm fully convinced that this helped much. If we have anyone who is seasoned on working with msbuild, I would love to get their input.

**Review by fmassa — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**